vividus-engine/src/main/java/org/vividus/monitor/ReplacementMonitor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Formatter;
+import java.util.Optional;
+
+import org.jbehave.core.steps.NullStepMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReplacementMonitor extends NullStepMonitor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReplacementMonitor.class);
+
+    private final ThreadLocal<ArrayList<String>> stepParametersCache = ThreadLocal.withInitial(ArrayList::new);
+
+    @Override
+    public void beforePerforming(String step, boolean dryRun, Method method)
+    {
+        getAnnotation(method).ifPresent(r -> {
+            try (Formatter newStep = new Formatter())
+            {
+                String[] stepParams = stepParametersCache.get().toArray(String[]::new);
+                newStep.format(r.replacementFormatPattern(), stepParams);
+
+                LOGGER.atInfo().addArgument(step).addArgument(newStep.toString())
+                        .log("The deprecated step \"{}\" was replaced by \"{}\"");
```

Comment:
how is this log info displayed in report?

Reply:
There was a problem with using "parametrized" version of step from monitor method
Raw step `beforeStep` looks well
Example for replacement pattern `When '%3$s' is '%2$s' '%1$s'`
![image](https://github.com/vividus-framework/vividus/assets/58571912/93523b4e-4f37-4459-8b12-cd64ba9fa94e)